clusterloader2/pkg/ticker/ticker.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ticker
+
+import (
+	"k8s.io/perf-tests/clusterloader2/api"
+)
+
+type ticker struct {
+	C     <-chan struct{}
+	timer timer
+}
+
+// NewTicker creates new ticker based on provided tuning set.
+func NewTicker(ts *api.TuningSet) *ticker {
+	ch := make(chan struct{})
+	return &ticker{
+		C:     ch,
+		timer: newTimer(ch, ts),
```

Comment:
why do you need to pass ch to newTimer when you already have it in member c?

Reply:
This way timer can have access to this channel.
`C` is read-only channel. The channel passed to `newTimer` is write-only.

With new tuning set strategies implemented I will rework this code, so strategies can register in Ticker. In this PR however I want only to focus on Ticker struct and methods, so it can be used in `ExecutePhase()` code.